### PR TITLE
fix: improve nested CSS rule handling and add related tests

### PR DESCRIPTION
**Pull request by alailsonko**
This pull request improves the handling and testing of nested CSS rule manipulations in rrweb. It fixes the traversal logic for nested CSS rules, enhances the utility function for retrieving nested rules, and adds comprehensive tests to ensure correctness when dealing with complex CSS structures (such as rules inside `@media` and `@supports` blocks). Additionally, new test events and integration tests are introduced to verify that style changes on nested rules are correctly replayed.

**Bug fixes and core logic improvements:**

* Fixed traversal in `getNestedCSSRulePositions` to ensure correct recursive parent rule lookup for nested CSS rules.
* Corrected the logic in `getNestedRule` to properly handle position arrays when accessing nested rules, ensuring accurate retrieval for any nesting depth.

**Testing enhancements:**

* Added a new test event file (`nested-style-declaration.ts`) that simulates style property changes (set/remove) on nested CSS rules, including deep nesting scenarios.
* Integrated these nested style declaration events into the main replay test suite, and added a comprehensive test that verifies rrweb's ability to replay style changes on nested rules at various points in time. [[1]](diffhunk://#diff-6687c9001cfae9096dcf767be76fbfb3ebb9f6726983d5ae2d67525092e49bacR28) [[2]](diffhunk://#diff-6687c9001cfae9096dcf767be76fbfb3ebb9f6726983d5ae2d67525092e49bacR1068-R1144)
* Expanded unit tests for the `getNestedRule` and `getPositionsAndIndex` utility functions to cover a wide range of nesting and edge cases, ensuring robustness of these utilities. [[1]](diffhunk://#diff-626495c150043789b5a8f567993dd33b5caa8b6a6e232e8d2b87fe4bd9c639faR10-R11) [[2]](diffhunk://#diff-626495c150043789b5a8f567993dd33b5caa8b6a6e232e8d2b87fe4bd9c639faR148-R251)

### CODE REVIEW

File: packages/rrweb/src/utils.ts [hunk 1]
```diff
@@ -420,18 +420,32 @@ export function hasShadowRoot<T extends Node | RRNode>(
...
+ * @param position - Array of indices, e.g., [0, 1, 0] for rules[0].cssRules[1].cssRules[0]
+ * @returns The nested rule, or null if not found
+ */
 export function getNestedRule(
   rules: CSSRuleList,
   position: number[],
-): CSSGroupingRule {
-  const rule = rules[position[0]] as CSSGroupingRule;
+): CSSGroupingRule | null {
+  const rule = rules?.[position[0]] as CSSGroupingRule;
```
Comment (Juice10) [inline packages/rrweb/src/utils.ts]:
```suggestion
  const rule = rules?.[position[0]] as CSSGroupingRule | null;
```

**Review by Juice10 — DISMISSED** [review]
